### ISSUE TITLE
OpenEOClient$connect ignores host set by OpenEOClient$new

### ISSUE THREAD

**Issue by benjaminschwetz** [Open-EO/openeo-r-client#100]
Hi, 

I just noticed that the `connect` method returns an error when I set the url via `new`. I tested with openeo 1.1.0

``` r
library(openeo)
#> 
#> Attaching package: 'openeo'
#> The following objects are masked from 'package:base':
#> 
#>     capabilities, debug
x = OpenEOClient$new(host = "example.foo")
x$connect()
#> Note: Host-URL is missing
```

<sup>Created on 2021-12-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by flahn** [comment]
Thanks for spotting this. The parameter `url` in `$connect()` was mandatory and did not take into account that it could have been set upon initialization. So, now you will receive a message, that the url *example.foo* cannot be resolved, what it is kind of to be expected.

**Comment by m-mohr** [comment]
Doesn't work for me yet, I'm getting:

```r
> library(openeo)
> x = OpenEOClient$new(host = "https://openeo.cloud")
> x$getHost()
[1] "https://openeo.cloud"
> x$connect()
Argument "url" fehlt (ohne Standardwert)
```

I'm on R 4.0.2

**Comment by m-mohr** [comment]
Weird, after updating to the latest commit I get:
```r
> library(openeo)
> x = OpenEOClient$new(host = "https://openeo.cloud")
> x$getHost()
[1] "https://openeo.cloud"
> x$connect()
host must be a single element of type 'character'
```

**Comment by m-mohr** [comment]
Fix for the issue above in PR #100
